src/impl/apis/maintenance_api_impl.py

```diff
@@ -341,20 +397,33 @@ async def delete_dataset(dataset: Dataset, pool: Pool = None) -> None:
         if result[0][0] > 0:
             sql = f"""
             DELETE FROM `{DB_NAME}`.`{table_name}`
-            WHERE `mdid` IN ({existing_mdids})
+            WHERE `mdid`={mdid})
             """
             await db.run(sql=sql, pool=pool)
-
+        logger.debug(
+            "Deleting metadata",
+            mdid=mdid,
+        )
         sql = f"""
```

Comment:
Not sure, but should the SQL live in `db.py`?

Reply:
There's a fair bit of raw SQL in this function, which I'm inclined to leave there for the time being pending a wider stratagem that changes how we handle the SQL/Schema, because at present there's a lot of slightly awkward code which is necessary to work around limitations in the code generation.